✨ HPC-8318: Allow for CDM frontend to handle multiple reporting windows

s0 opened the pull request.

s0 committed:
Allow for operations with multiple reporting windows

Automatically navigate to the most appropriate reporting window,
picking open windows first.

s0 committed:
✨ HPC-8318: Allow switching between reporting windows

Use the sidebar component to display the reporting windows for any
given operation, when listing an operation's or cluster's forms,
and allow for switching between them.

s0 committed:
✨ HPC-8318: Display reporting window in form data breadcrumbs

When displaying the form data for a particular form,
include the reporting window name in the breadcrumbs that are displayed
above the form.

s0 committed:
🎨 Adjust margins / padding / height for sidenav changes

To allow for better alignment of sidebar navigation and existing
elements on pages, adjust the padding / margins across a number of
elements for more consistent spacing.

In particular, make list headers, list items and sidebar nav buttons the
same height.

s0 committed:
⬆️ Update react-icons for latest icons

s0 committed:
✨ HPC-8318: Add state icons to reporting windows

Add:

* The capability to use Icons in sidebar navigation
* Usage of icons for reporting window lists
* Improved styling to handle long reporting window names

s0 committed:
✨ HPC-8318: Simplify assignment state and remove canModifyWhenClean

To simplify assignment permissions, and modify the behaviour of
`editable` to indicate if the user has permission to edit the form,
we've removed `canModifyWhenClean`, and modify the UI display based
solely on assignment state and the `editable` flag.

The form state message has also been updated to reflect to the user
when inability to edit the form may be down to the reporting window
state.

czmj reviewed (changes requested):
@s0 `FormToolbar` needs to pass a reportingWindow to the props for `Indicator` now (which is why CI is failing) but I'm happy with everything else

s0 committed:
🔧  Run npm run build:affected in pre-commit

To catch type-errors and build-errors (in addition to lint errors),
run build:affected in the pre-commit check.

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

s0 commented:
@czmj addressed, and also updated the pre-commit hook to also run a build so these things are caught before PRs going forward :-D
